### ISSUE TITLE
SelectControl custom classes applied incorrectly

### ISSUE THREAD

**Issue by picocodes** [WordPress/gutenberg#50359]
### Description

Other input controls apply the custom CSS classes to the parent BaseControl component. SelectControl, on the other hand, applies the custom CSS classes to the inner InputBase component.

This makes it impossible to apply custom CSS styles.

### Step-by-step reproduction instructions

Check https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/select-control/index.tsx#L98

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mirka** [comment]
Oof, this is indeed a mistake, and something that cannot be easily fixed in a backward compatible way 😅

I can only suggest things like using a wrapper div, or a parent selector like `.components-base-control:has(.my-select-control-class)`.

**Comment by jorgefilipecosta** [comment]
Hi @mirka, do you think adding a new class prop like containerClass, baseClass could be a potential way to fix this issue? Otherwise I guess we can close it as something that could not be fixed now.

**Comment by mirka** [comment]
Yes, I guess we can close as a won't fix since we have a workaround, and it will be fixed soon in the new version of `SelectControl`.
